File: C++/minimum-add-to-make-parentheses-valid.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int balance = 0, mustOpen = 0;
        for (char c: s) {
            switch(c) {
                case '(': balance++; break;
                case ')': balance--; break;
            }
            if (balance == -1) {
                balance = 0;
                mustOpen++;
            }
        }
        return mustOpen + balance;
    }
};

File: Go/minimum-add-to-make-parentheses-valid.go
func minAddToMakeValid(s string) int {
	balance, mustOpen := 0, 0
	for _, c := range s {
		if c == '(' {
			balance++
		} else if balance > 0 {
			balance--
		} else {
			mustOpen++
		}
	}
	return balance + mustOpen
}